server/plugin_test.go

```diff
@@ -91,9 +91,9 @@ func TestPlugin(t *testing.T) {
 		},
 		"CreatePostError": {
 			Configuration:      validConfiguration,
-			CreatePostError:    model.NewAppError("foo", "bar", nil, "", http.StatusInternalServerError),
+			CreatePostError:    model.NewAppError("foo", "bar", nil, "", http.StatusBadRequest),
 			Request:            httptest.NewRequest("POST", "/webhook?team=theteam&channel=thechannel&secret=thesecret", validRequestBody()),
-			ExpectedStatusCode: http.StatusInternalServerError,
+			ExpectedStatusCode: http.StatusBadRequest,
```

Comment:
Why was there a behaviour change?

Reply:
Earlier, we were returning the status Code from the appError. However, since the error no longer has access to the returned Status Code, I had hardcoded it as `http.StatusBadRequest` which had caused this change. I will revert and hardcode the return status as `http.StatusInternalServerError` instead.

<img width="1311" alt="Screenshot 2023-03-24 at 10 45 21 AM" src="https://user-images.githubusercontent.com/1429138/227430828-b1641bec-03bd-434c-8e01-7986bbe8c0f3.png">